src/cascadia/TerminalSettingsEditor/ProfileViewModel.h

```diff
@@ -26,6 +27,22 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         Editor::ProfileViewModel _Profile{ nullptr };
     };
 
+    struct BellSoundViewModel : BellSoundViewModelT<BellSoundViewModel>, ViewModelHelper<BellSoundViewModel>
+    {
+    public:
+        BellSoundViewModel(hstring path, const Windows::UI::Core::CoreDispatcher& dispatcher);
+
+        hstring DisplayPath() const;
+        hstring SubText() const;
+        VIEW_MODEL_OBSERVABLE_PROPERTY(bool, FileExists, true);

```

Comment:
technically, if you want to do this right - `FileExists` changing results in `DisplayPath` and `SubText` changing too. You can make them observable just by Notifying with their names - and XAML will automatically follow.

This would also provide some validation that your coroutine is working. Right now, DisplayPath and SubText MIGHT be getting queried before FileExists is set (!!). This will be especially true if one of the bell sound files is on a network share.

You can simulate this by adding a `Sleep(2000)` to the background portion of the per-bell-sound coroutiune.